src/bin/cargo/commands/install.rs

```diff
@@ -108,6 +109,18 @@ pub fn exec(config: &mut Config, args: &ArgMatches) -> CliResult {
         .map(|k| resolve_crate(k, version))
         .collect::<crate::CargoResult<Vec<_>>>()?;
 
+    for krate in krates.iter() {
+        let crate_name = krate.0;
+        if !crate_name.starts_with("+") {
+            continue;
+        }
+
+        let crate_name_wo_prefix = crate_name[1..].to_string();
+
+        return Err(anyhow!("invalid package name: \"{crate_name}\"
+    Did you meant to use a toolchain? Use `cargo {crate_name} install` instead to use the `{crate_name_wo_prefix}` toolchain.").into());
```

Comment:
https://rustc-dev-guide.rust-lang.org/diagnostics.html#suggestion-style-guide

Although Cargo still uses style like “Did you mean”, I personally try to avoid it. See the link above.

Also, `+` prefixed name is already an invalid crate name. We don't need to question users again. They must already do something wrong.

Here is my old shorter suggestion for comparsion:

```suggestion
    Use `cargo {crate_name} install` if you meant to use the `{crate_name_wo_prefix}` toolchain.").into());
```

Reply:
I can't come up with something better than your version for the second line, so I'll just use that instead.